packages/blade/src/utils/useInterval.ts

```diff
@@ -0,0 +1,30 @@
+/* eslint-disable consistent-return */
+import React from 'react';
+import { useIsomorphicLayoutEffect } from './useIsomorphicLayoutEffect';
+
+function useInterval(
+  callback: () => void,
+  { delay, enable }: { delay: number; enable?: boolean },
+): void {
+  const intervalRef = React.useRef<number | null>(null);
+  const savedCallback = React.useRef(callback);
+
+  // keep the callback updated
+  useIsomorphicLayoutEffect(() => {
+    savedCallback.current = callback;
+  }, [callback]);
+
+  React.useEffect(() => {
```

Comment:
what exactly does this hook do?

Reply:
Same as `window.setInterval(() => {}, 5000)`